storage/replica.go

```diff
@@ -1559,6 +1559,10 @@ func (r *Replica) addWriteCmd(
 				} else {
 					log.Warningf(ctx, "unable to cancel expired Raft command %s", ba)
 				}
+			case <-r.store.stopper.ShouldQuiesce():
```

Comment:
This will actually have the request spin hot on this code path if `tryAbandon` returns false, but if that happens the command should be executing and finish momentarily. I think we can risk this for now (or you would have to follow that same pattern as with `ctxDone` above).